pkg/tests/cross-tests/rapid_tv_gen.go

```diff
@@ -178,15 +178,15 @@ func (tvg *tvGen) GenSetAttr(depth int) *rapid.Generator[tv] {
 }
 
 // TF blocks can be resource or datasource inputs, or nested blocks.
-func (tvg *tvGen) GenBlockWithDepth(depth int) *rapid.Generator[tb] {
+func (tvg *tvGen) GenBlockWithDepth(depth int, parentName string) *rapid.Generator[tb] {
 	return rapid.Custom[tb](func(t *rapid.T) tb {
 		fieldSchemas := map[string]*schema.Schema{}
 		fieldTypes := map[string]tftypes.Type{}
 		fieldGenerators := map[string]*rapid.Generator[tftypes.Value]{}
 		nFields := rapid.IntRange(0, 3).Draw(t, "nFields")
 		for i := 0; i < nFields; i++ {
-			fieldName := fmt.Sprintf("d%df%d", depth, i)
-			fieldTV := tvg.GenBlockOrAttrWithDepth(depth-1).Draw(t, fieldName)
+			fieldName := fmt.Sprintf("%sd%df%d", parentName, depth, i)
```

Comment:
Ah looks like I missed that fieldName is part of the schema e.g. through fieldTypes. I think this probably depends on what are you trying to test out. Originally writing the generators I wasn't particularly interested in testing support for various property naming issues like nested repetition or pluralization, therefore name generator was intentionally kept spartan and low-entropy to not waste entropy on the names. Looks like this optimizes it for debug-ability while keeping it low-entropy. IF there's no one-size-fits all strategy for generated prop names this can become a param.